src/shared/container/jobs/MailProviderJob/fakes/FakeMailProviderJob.ts

```diff
@@ -0,0 +1,29 @@
+import IJob, { IJobData } from '@shared/container/jobs/interfaces/IJob';
+import ISendMailDTO from '@shared/container/providers/MailProvider/dtos/ISendMailDTO';
+import IMailProvider from '@shared/container/providers/MailProvider/interfaces/IMailProvider';
+import FakeMailProvider from '@shared/container/providers/MailProvider/fakes/FakeMailProvider';
+import { MAIL_PROVIDER_JOB_KEY } from '@shared/container/jobs/keys';
+
+class FakeMailProviderJob implements IJob {
+  private emailProvider: IMailProvider;
+
+  constructor() {
+    this.emailProvider = new FakeMailProvider();
+  }
+
+  get key(): string {
+    return MAIL_PROVIDER_JOB_KEY;
+  }
+
+  public async handle(job: IJobData<ISendMailDTO>): Promise<void> {
```

Comment:
As this is an async method and it doesn't have a return statement, you really don't need the `Promise<void>`. TypeScript infers that it will be a promise with no return value. 

```typescript
public async handle(job: IJobData<ISendMailDTO>) {
```

Letting the TS compiler infers types for us make our code cleaner.

Reply:
Nice catch!